UnhandledPromiseRejectionWarning: #<Object> warning

Getting following warning on calling generateSession function with invalid code.

```
(node:5690) UnhandledPromiseRejectionWarning: #<Object>
(Use `node --trace-warnings ...` to show where the warning was created)
(node:5690) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 3)
(node:5690) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Here is [runkit link](https://runkit.com/sunilgautam/5f7ab7e7972ffa001ba1be5e) to reproduce this issue.

@sanchitrk 
i have created .npmrc file and tried to install the same, but ran into problem, below is error log, Kindly help me to resolve this.


D:\WORK\FIRETICK>npm install @socialstockdev/kiteconnect@3.1.3
npm ERR! code E401
npm ERR! Unable to authenticate, need: Basic realm="GitHub Package Registry"

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Arjun\AppData\Roaming\npm-cache\_logs\2021-02-11T04_42_57_553Z-debug.log

--- a/lib/connect.js
+++ b/lib/connect.js
@@ -434,21 +434,24 @@ var KiteConnect = function(params) {
      * @param {string} api_secret API secret issued with the API key.
      */
     self.generateSession = function(request_token, api_secret) {
-        var checksum = sha256(self.api_key + request_token + api_secret).toString();
+        return new Promise(function (resolve, reject) {
+            var checksum = sha256(self.api_key + request_token + api_secret).toString();
+            var p = _post("api.token", {
+                    api_key: self.api_key,
+                    request_token: request_token,
+                    checksum: checksum
+                }, null, formatGenerateSession);
 
-        var p = _post("api.token", {
-                api_key: self.api_key,
-                request_token: request_token,
-                checksum: checksum
-            }, null, formatGenerateSession);
-
-        p.then(function(response) {
-            self.setAccessToken(response.access_token);
-        }).catch(function(err) {
-            throw err
+            p.then(function(resp) {
+                // Set access token.
+                if (resp && resp.access_token) {
+                    self.setAccessToken(resp.access_token);
+                }
+                return resolve(resp);
+            }).catch(function(err) {
+                return reject(err);
+            });
         });
-
-        return p;
     };
 
     /**
@@ -479,21 +482,24 @@ var KiteConnect = function(params) {
      * @param {string} api_secret API secret issued with the API key.
      */
     self.renewAccessToken = function(refresh_token, api_secret) {
-        var checksum = sha256(self.api_key + refresh_token + api_secret).toString();
+        return new Promise(function (resolve, reject) {
+            var checksum = sha256(self.api_key + refresh_token + api_secret).toString();
 
-        var p = _post("api.token.renew", {
-            api_key: self.api_key,
-            refresh_token: refresh_token,
-            checksum: checksum
+            var p = _post("api.token.renew", {
+                api_key: self.api_key,
+                refresh_token: refresh_token,
+                checksum: checksum
+            });
+
+            p.then(function(resp) {
+                if (resp && resp.access_token) {
+                    self.setAccessToken(resp.access_token);
+                }
+                return resolve(resp);
+            }).catch(function(err) {
+                return reject(err);
+            });
         });
-
-        p.then(function(response) {
-            self.setAccessToken(response.access_token);
-        }).catch(function(err) {
-            throw err
-        });
-
-        return p;
     };
 
     /**
@@ -668,7 +674,7 @@ regular).
      * @param {Object} params   Margin fetch order params.
      * @param {string} params.exchange    Name of the exchange(eg. NSE, BSE, NFO, CDS, MCX)
      * @param {string} params.tradingsymbol   Trading symbol of the instrument
-     * @param {string} params.transaction_type   eg. BUY, SELL  
+     * @param {string} params.transaction_type   eg. BUY, SELL
      * @param {string} params.variety   Order variety (regular, amo, bo, co etc.)
      * @param {string} params.product   Margin product to use for the order
      * @param {string} params.order_type   Order type (MARKET, LIMIT etc.)
@@ -1300,12 +1306,12 @@ regular).
         let queryParams = null;
         if (method === "GET" || method === "DELETE") {
             queryParams = params;
-        } else { 
+        } else {
             if (isJSON) {
             // post JSON payload
             payload = JSON.stringify(params);
             } else {
-            // post url encoded payload    
+            // post url encoded payload
             payload = querystring.stringify(params);
             }
         }
@@ -1330,7 +1336,7 @@ regular).
             options["headers"]["Content-Type"] = "application/json"
         } else {
             options["headers"]["Content-Type"] = "application/x-www-form-urlencoded"
-        } 
+        }
         // Set response transformer
         if (responseTransformer) {
             options.transformResponse = axios.defaults.transformResponse.concat(responseTransformer);

--- a/lib/connect.js
+++ b/lib/connect.js
@@ -1021,13 +1021,13 @@ regular).
 	// Get API params from user defined GTT params.
     self._getGTTPayload = function (params) {
         if (params.trigger_type !== self.GTT_TYPE_OCO && params.trigger_type !== self.GTT_TYPE_SINGLE) {
-            throw("Invalid `params.trigger_type`")
+            throw new Error("Invalid `params.trigger_type`")
         }
         if (params.trigger_type === self.GTT_TYPE_OCO && params.trigger_values.length !== 2) {
-            throw ("Invalid `trigger_values` for `OCO` order type")
+            throw new Error("Invalid `trigger_values` for `OCO` order type")
         }
         if (params.trigger_type === self.GTT_TYPE_SINGLE && params.trigger_values.length !== 1) {
-            throw ("Invalid `trigger_values` for `single` order type")
+            throw new Error("Invalid `trigger_values` for `single` order type")
         }
         let condition = {
             exchange: params.exchange,

--- a/package.json
+++ b/package.json
@@ -27,10 +27,10 @@
   },
   "homepage": "https://github.com/zerodhatech/kiteconnectjs#readme",
   "dependencies": {
-    "axios": "^0.17.1",
+    "axios": "^0.21.1",
     "crypto-js": "^3.1.9-1",
-    "papaparse": "^4.3.6",
-    "ws": "^1.1.1"
+    "papaparse": "^5.2.0",
+    "ws": "^3.3.1"
   },
   "devDependencies": {
     "jsdoc": "^3.4.0",

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "kiteconnect",
-  "version": "3.1.0",
+  "version": "3.2.0",
   "description": "The official JS client library for the Kite Connect trading APIs",
   "main": "./lib/",
   "scripts": {

